lib/RecommendationProvider.js

```diff
@@ -48,13 +48,13 @@ RecommendationProvider.prototype = {
     * Handle the NEW_RECOMMENDATION event
     */
   _handleNewRecommendationEvent(currentRecommendation) {
-    let url = currentRecommendation ? currentRecommendation.url : null;
-    let recommenderType = currentRecommendation ? currentRecommendation.recommender_type : null;
-    this._tabTracker.handleUserEvent({
-      event: "NEW_RECOMMENDATION",
-      url: url,
-      recommender_type: recommenderType
-    });
+    if (currentRecommendation && currentRecommendation.url &&
+      currentRecommendation.recommender_type) {
+      let payload = {event: "NEW_RECOMMENDATION"};
+      payload.url = currentRecommendation.url;
+      payload.recommender_type = currentRecommendation.recommender_type;
```

Comment:
In the unit test [here](https://github.com/mozilla/activity-stream/blob/master/test/test-RecommendationProvider.js#L88), there is a special case where the `currentRecommendation` is `null`. So I guess if the `currentRecommendation` is null, we shuoldn't send out a ping with both "url" and "recommender_type" as null.

Does it make sense?